src/v/debug_bundle/types.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "types.h"
+
+#include <seastar/util/variant_utils.hh>
+
+#include <fmt/chrono.h>
+
+auto fmt::formatter<debug_bundle::special_date>::format(
+  debug_bundle::special_date d, format_context& ctx) const
+  -> format_context::iterator {
+    return formatter<string_view>::format(debug_bundle::to_string_view(d), ctx);
+}
+
+auto fmt::formatter<debug_bundle::time_variant>::format(
+  const debug_bundle::time_variant& t, format_context& ctx) const
+  -> format_context::iterator {
+    return ss::visit(
+      t,
+      [&ctx](const debug_bundle::clock::time_point& t) {
+          auto tt = debug_bundle::clock::to_time_t(t);
+          std::tm tm = *std::localtime(&tt);
```

Comment:
not to belabor this too much, but is this actually desirable? what if I'm kicking off debug bundles on brokers in different time zones?

Reply:
Can we just use unix timestamps everywhere in the API contracts and just let RPK handle the timezone conversion. Timezones are a mess and if we can isolate that to a single place that's awesome.

Reply:
My intention (maybe it's wrong) is to try to make things easier for the user.  They can see the timestamps in their logs (which if they're collected via journal will be in the local time zone) and be able to select the start/stop times they want to bound their log message collection.  Rather than forcing them to have to convert local time to UTC when providing the start/stop time options.

Reply:
> let RPK handle the timezone conversion

I don't think these are argument necessarily for RPK but are actually passed to journal

Reply:
>> let RPK handle the timezone conversion

> I don't think these are argument necessarily for RPK but are actually passed to journal

Two scenarios:
* bare-metal deployments, these arguments are passed directly to journal
* k8s - looks like it treats the time as the local time (no tz conversion appears done)

Reply:
>rather than forcing them to ... convert local time to UTC

Still don't understand where the conversion will happen though if end user and broker are in different time zones themselves. If the API contract is "UNIX timestamp", then there's no ambiguity anywhere, right?

Reply:
My basic motivation for all of the options is to provide a very similar UX for a user familiar with running `rpk debug bundle` from the command line.

Reply:
> end user and broker are in different time zones themselves

I'd imagine the end user will provide timestamps in the same tz as the broker since the broker logs will report with timestamps in that tz.